core/src/main/java/org/elasticsearch/discovery/zen/elect/ElectMasterService.java

```diff
@@ -41,23 +40,7 @@
  */
 public class ElectMasterService extends AbstractComponent {
 
-    public static final String DISCOVERY_ZEN_MINIMUM_MASTER_NODES = "discovery.zen.minimum_master_nodes";
-    public static final Validator DISCOVERY_ZEN_MINIMUM_MASTER_NODES_VALIDATOR = new Validator() {
-        @Override
-        public String validate(String setting, String value, ClusterState clusterState) {
-            int intValue;
-            try {
-                intValue = Integer.parseInt(value);
-            } catch (NumberFormatException ex) {
-                return "cannot parse value [" + value + "] as an integer";
-            }
-            int masterNodes = clusterState.nodes().masterNodes().size();
-            if (intValue > masterNodes) {
-                return "cannot set " + ElectMasterService.DISCOVERY_ZEN_MINIMUM_MASTER_NODES + " to more than the current master nodes count [" + masterNodes + "]";
-            }
-            return null;
-        }
-    };
+    public static final Setting<Integer> DISCOVERY_ZEN_MINIMUM_MASTER_NODES_SETTING = Setting.intSetting("discovery.zen.minimum_master_nodes", -1, true, Setting.Scope.Cluster);
```

Comment:
that is only validated on update. - not a bug, that's a feature. you can set min master nodes to a value higher than the number of nodes but only in your yaml file. :dancer: not on update.